utils/contracts.ts

```diff
@@ -1,46 +1,118 @@
 import { Contract } from 'ethers';
+import type {
+  BridgedPolygonNORI,
+  Certificate,
+  FIFOMarket,
+  LockedNORI,
+  NORI,
+  Removal,
+} from '../typechain-types';
 
-import type { BridgedPolygonNORI, LockedNORI } from '../typechain-types';
-import type { networks } from '../config/networks';
+export const getContract = async ({
+  contractName,
+  hre,
+  signer,
+}: {
+  contractName: string;
+  hre: CustomHardHatRuntimeEnvironment;
+  signer?: ConstructorParameters<typeof Contract>[2];
+}): Promise<Contract> => {
+  const contract = await hre.ethers.getContractAt(
+    contractName,
+    (
+      await hre.deployments.get(contractName)
+    ).address
+  );
+  if (!contract) {
+    throw new Error(`Unsupported network: ${hre.network.name}`);
+  }
+  if (signer) {
+    return contract.connect(signer);
+  }
+  return contract;
+};
 
-import * as contractsConfig from '@/contracts.json';
-import { abi as bridgedPolygonNoriAbi } from '@/artifacts/BridgedPolygonNORI.sol/BridgedPolygonNORI.json';
-import { abi as lockedNoriAbi } from '@/artifacts/LockedNORI.sol/LockedNORI.json';
+// TODO Is there a smarter way to do this typing dance?
```

Comment:
```ts
export const getContract = async <
  TContract extends ContractInstances[keyof ContractInstances]
>({
  contractName,
  hre,
  signer,
}: {
  contractName: TContract extends BridgedPolygonNORI
    ? 'BridgedPolygonNORI'
    : TContract extends LockedNORI
    ? 'LockedNORI'
    : TContract extends NORI
    ? 'NORI'
    : TContract extends Removal
    ? 'Removal'
    : TContract extends Certificate
    ? 'Certificate'
    : TContract extends FIFOMarket
    ? 'FIFOMarket'
    : never;
  hre: CustomHardHatRuntimeEnvironment;
  signer?: ConstructorParameters<typeof Contract>[2];
}): Promise<TContract> => {
  const contract = await hre.ethers.getContractAt(
    contractName,
    (
      await hre.deployments.get(contractName)
    ).address
  );
  if (!contract) {
    throw new Error(`Unsupported network: ${hre.network.name}`);
  }
  return (signer != null ? contract.connect(signer) : contract) as TContract;
};
```

Reply:
Thanks! Any way to avoid needing that giant ternary chain?